bascule/basculehttp/constructor.go

```diff
@@ -15,18 +17,23 @@ const (
 type constructor struct {
 	headerName     string
 	authorizations map[bascule.Authorization]TokenFactory
+	getLogger      func(context.Context) bascule.Logger
 }
 
 func (c *constructor) decorate(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+		logger := c.getLogger(request.Context())
```

Comment:
This is the right approach.  You just need to allow for the case when `getLogger` is unset, which means the client of `bascule` didn't configure it.  You can use the `go-kit` `NopLogger` as a fallback if `getLogger == nil`.  That way, you don't have to check for `nil` all over the place.